src/com/amazon/ionelement/api/AnyElement.kt

```diff
@@ -149,207 +149,411 @@ import java.math.BigInteger
+     * If this element is not an Ion `list`, `sexp`, or `struct, or if it is any Ion `null`, throws an
+     * [IonElementConstraintException].
+     */
     public fun asContainer(): ContainerElement
 
-    /** See [AnyElement]. */
+    /**
+     * Attempts to narrow this element to a [ContainerElement] or `null`.
+     * If this element is not an Ion `list`, `sexp`, `struct`, or `null.null`, throws an [IonElementConstraintException].
+     */
     public fun asContainerOrNull(): ContainerElement?
 
-    /** See [AnyElement]. */
+    /**
+     * Attempts to narrow this element to a [SeqElement].
+     * If this element is not an Ion `list` or `sexp`, or if it is any Ion `null`, throws an
+     * [IonElementConstraintException].
+     */
     public fun asSeq(): SeqElement
 
-    /** See [AnyElement]. */
+    /**
+     * Attempts to narrow this element to a [SeqElement] or `null`.
+     * If this element is not an Ion `list`, `sexp`, or `null.null`, throws an [IonElementConstraintException].
+     */
     public fun asSeqOrNull(): SeqElement?
 
-    /** See [AnyElement]. */
+    /**
+     * Attempts to narrow this element to a [ListElement].
+     * If this element is not an Ion `list`, or if it is `null.list`, throws an [IonElementConstraintException].
+     */
     public fun asList(): ListElement
 
-    /** See [AnyElement]. */
+    /**
+     * Attempts to narrow this element to a [ListElement] or `null`.
+     * If this element is not an Ion `list` or `null.null`, throws an [IonElementConstraintException].
+     */
     public fun asListOrNull(): ListElement?
 
-    /** See [AnyElement]. */
+    /**
+     * Attempts to narrow this element to a [SexpElement].
+     * If this element is not an Ion `sexp`, or if it is `null.sexp`, throws an [IonElementConstraintException].
+     */
     public fun asSexp(): SexpElement
 
-    /** See [AnyElement]. */
+    /**
+     * Attempts to narrow this element to a [SexpElement] or `null`.
+     * If this element is not an Ion `sexp` or `null.null`, throws an [IonElementConstraintException].
+     */
     public fun asSexpOrNull(): SexpElement?
 
-    /** See [AnyElement]. */
+    /**
+     * Attempts to narrow this element to a [StructElement].
+     * If this element is not an Ion `struct`, or if it is `null.struct`, throws an [IonElementConstraintException].
+     */
     public fun asStruct(): StructElement
 
-    /** See [AnyElement]. */
+    /**
+     * Attempts to narrow this element to a [StructElement] or `null`.
+     * If this element is not an Ion `struct` or `null.null`, throws an [IonElementConstraintException].
+     */
     public fun asStructOrNull(): StructElement?
 
     /**
      * If this is an Ion integer, returns its [IntElementSize] otherwise, throws [IonElementConstraintException].
      */
     public val integerSize: IntElementSize
 
-    /** See [AnyElement]. */
+    /**
+     * Gets the value of the element, assuming that it is a non-null Ion `bool`.
```

Comment:
Nit: I wouldn't use the term 'gets' for properties.

Reply:
I was not thrilled with "gets", but I couldn't think of a better way to phrase it. Got any suggestions?

Reply:
What about something like this?
```suggestion
     * The value of the element, assuming that the element is a non-null Ion `bool`.
     * Attempting to access this property on an element that violates the type assumption will result in an  [IonElementConstraintException].
```

Reply:
Yea... I want to suggest "Access" though that may be a bit wordy.